src/Language/PureScript/TypeChecker.hs

```diff
@@ -727,3 +730,23 @@ typeCheckModule (Module ss coms mn decls (Just exps)) =
     extractMemberName (TypeDeclaration td) = tydeclIdent td
     extractMemberName _ = internalError "Unexpected declaration in typeclass member list"
   checkClassMembersAreExported _ = return ()
+
+  checkDataConstructorsAreExported :: DeclarationRef -> m ()
+  checkDataConstructorsAreExported dr@(TypeRef ss' name exportedDataConstructorsNames)
+    | null $ fromMaybe [] exportedDataConstructorsNames = for_
+      [ DataNewtype.Newtype
+      , DataGenericRep.Generic
```

Comment:
I introduced `Language.PureScript.Constants.Data.Generic.Rep` and `Language.PureScript.Constants.Data.Newtype` in order to not duplicate the names of those classes here and during type classes instances deriving but perhaps it would be ok to repeat them?